File: src/Entity/Commentaire.php
<?php

namespace App\Entity;

use App\Repository\CommentaireRepository;
use Doctrine\ORM\Mapping as ORM;
use JsonSerializable;
use Symfony\Component\Serializer\Annotation\Groups;

/**
 * @ORM\Entity(repositoryClass=CommentaireRepository::class)
 */
class Commentaire implements JsonSerializable
{
    /**
     * @ORM\Id
     * @ORM\GeneratedValue
     * @ORM\Column(type="integer")
     * @Groups("comment")
     */
    private $id;

    /**
     * @ORM\Column(type="string", length=255)
     * @Groups("comment")
     */
    private $comment;

    /**
     * @ORM\ManyToOne(targetEntity=Maison::class, inversedBy="commentaires")
     * @ORM\JoinColumn(nullable=false)
     * @Groups("comment")
     */
    private $maison;

    /**
     * @ORM\ManyToOne(targetEntity=User::class, inversedBy="commentaires")
     * @ORM\JoinColumn(nullable=false)
     * @Groups("comment")
     */
    private $user;

    /**
     * @ORM\Column(type="date")
     * @Groups("comment")
     */
    private $date;


    /**
     * Commentaire constructor.
     * @param $date
     */

    public function __construct(){

        $this->date = new \DateTime();
    }

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getComment(): ?string
    {
        return $this->comment;
    }

    public function setComment(string $comment): self
    {
        $this->comment = $comment;

        return $this;
    }

    public function getMaison(): ?Maison
    {
        return $this->maison;
    }

    public function setMaison(?Maison $maison): self
    {
        $this->maison = $maison;

        return $this;
    }

    public function getUser(): ?User
    {
        return $this->user;
    }

    public function setUser(?User $user): self
    {
        $this->user = $user;

        return $this;
    }

    public function getDate(): ?\DateTimeInterface
    {
        return $this->date;
    }

    public function setDate(\DateTimeInterface $date): self
    {
        $this->date = $date;

        return $this;
    }

    public function jsonSerialize()
    {
        return array(
            'id' => $this->id,
            'comment' => $this->comment,
            'date' => $this->date,

        );
    }
}

File: Tfarhida/src/Tfarhida/entities/Commentaire.java
package Tfarhida.entities;

import java.sql.Date;


public class Commentaire {
    private int id, maison_id, user_id;
    private String comment;
    private Date date;

    public Commentaire(){
        
    }
    public Commentaire(int id, int maison_id, int user_id, String comment, Date date) {
        this.id = id;
        this.maison_id = maison_id;
        this.user_id = user_id;
        this.comment = comment;
        this.date = date;
    }
    
    public Commentaire(int maison_id, int user_id, String comment, Date date) {
        this.maison_id = maison_id;
        this.user_id = user_id;
        this.comment = comment;
        this.date = date;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public int getMaison_id() {
        return maison_id;
    }

    public void setMaison_id(int maison_id) {
        this.maison_id = maison_id;
    }

    public int getUser_id() {
        return user_id;
    }

    public void setUser_id(int user_id) {
        this.user_id = user_id;
    }

    public String getComment() {
        return comment;
    }

    public void setComment(String comment) {
        this.comment = comment;
    }

    public Date getDate() {
        return date;
    }

    public void setDate(Date date) {
        this.date = date;
    }

    @Override
    public String toString() {
        return "Commentaire{" + "id=" + id + ", maison_id=" + maison_id + ", user_id=" + user_id + ", comment=" + comment + ", date=" + date + '}';
    }
    
    
    
}
